client-go/controller/client/auth.go

```diff
@@ -0,0 +1,243 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/deis/deis/client-go/controller/api"
+)
+
+// Register a user with the controller
+func Register(URL url.URL, username string, password string, email string, sslVerify bool,
+	loginAfter bool) error {
+	controllerURL := URL
+	client := CreateHTTPClient(sslVerify)
+
+	// Create a JSON body of request from user api
+	user := api.AuthRegisterRequest{Username: username, Password: password, Email: email}
+	body, err := json.Marshal(user)
+
+	if err != nil {
+		return err
+	}
+
+	URL.Path = "/v1/auth/register/"
+
+	headers := http.Header{}
+
+	controllerClient, err := Load()
+
+	if err == nil {
+		headers.Add("Authorization", "token "+controllerClient.Token)
+	}
+
+	headers.Add("Content-Type", "application/json")
+	addUserAgent(&headers)
+
+	res, err := rawRequest(client, "POST", URL.String(), bytes.NewBuffer(body), headers, 201)
+
+	if err != nil {
+		return err
+	}
+
+	res.Body.Close()
+
+	fmt.Printf("Registered %s\n", username)
+
+	if loginAfter {
+		return Login(controllerURL, username, password, sslVerify)
+	}
+
+	return nil
+}
+
+// Login to the controller
+func Login(URL url.URL, username string, password string, sslVerify bool) error {
+	client := CreateHTTPClient(sslVerify)
+
+	controllerURL := URL
+
+	// Create a JSON body of request from api
+	user := api.AuthLoginRequest{Username: username, Password: password}
+	body, err := json.Marshal(user)
+
+	if err != nil {
+		return err
+	}
+
+	URL.Path = "/v1/auth/login/"
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json")
+	addUserAgent(&headers)
+
+	res, err := rawRequest(client, "POST", URL.String(), bytes.NewBuffer(body), headers, 200)
+
+	if err != nil {
+		return err
+	}
+
+	resBody, err := ioutil.ReadAll(res.Body)
+
+	token := api.AuthLoginResponse{}
+
+	err = json.Unmarshal([]byte(resBody), &token)
+	res.Body.Close()
+
+	if err != nil {
+		return err
+	}
+
+	controllerClient := Client{Username: username, SSLVerify: sslVerify,
+		URL: controllerURL, Token: token.Token}
+
+	err = controllerClient.Save()
+
+	if err != nil {
+		return nil
```

Comment:
Should be `return err` I assume (and I would make it a one-line check with `controllerClient.Save()` above).